GitTfs/Core/TfsInterop/RootBranch.cs

```diff
@@ -5,19 +5,36 @@ namespace Sep.Git.Tfs.Core.TfsInterop
     [DebuggerDisplay("{DebuggerDisplay}")]
     public class RootBranch
     {
-        public RootBranch(int rootChangeset, string tfsBranchPath)
+        public string TfsBranchPath { get; }
+
+        public int SourceBranchChangesetId { get; }
+
+        public int TargetBranchChangesetId { get; }
+
+        public RootBranch(int sourceBranchChangesetId, string tfsBranchPath)
+            : this(sourceBranchChangesetId, -1, tfsBranchPath)
+        {
+
+        }
+
+        public RootBranch(int sourceBranchChangesetId, int targetBranchChangesetId, string tfsBranchPath)
         {
-            RootChangeset = rootChangeset;
+            SourceBranchChangesetId = sourceBranchChangesetId;
+            TargetBranchChangesetId = targetBranchChangesetId;
             TfsBranchPath = tfsBranchPath;
         }
 
-        public int RootChangeset { get; private set; }
-        public string TfsBranchPath { get; private set; }
         public bool IsRenamedBranch { get; set; }
 
         private string DebuggerDisplay
         {
-            get { return string.Format("{0} C{1}{2}", TfsBranchPath, RootChangeset, (IsRenamedBranch ? " renamed" : "")); }
+            get
+            {
+                if (TargetBranchChangesetId > -1)
```

Comment:
I think the chance of this logic being needed elsewhere is slim. On one hand, I agree that a property shouldn't do computation or logic. On the other hand, we'd then be adding a private method for the sole purpose of building a string exposed by a property. They feel equally wrong to me.